### PR TITLE
Hsd8 531 refactor admin theme

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Uninstall material admin theme and disable theme switcher module

# Needed By (Date)
- End of sprint 1/30

# Urgency
- high

# Steps to Test
1. checkout this branch
1. sync a sites `blt drupal:sync --site=[sitename]`
1. ensure admin theme looks like seven. 
1. validate on the appearance page, `Humsci Admin` is selected as the admin theme.
1. validate all content still appears on the page.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 1]
```diff
@@ -283,3 +283,36 @@ function _su_humsci_profile_update_fix_menuposition_sections(array $sections) {
...
 }
+
+/**
+ * Uninstall material admin theme and theme switcher module.
+ */
+function su_humsci_profile_update_8012() {
+  $config_factory = \Drupal::configFactory();
+
+  // Delete the old block configurations for the admin theme so that when the
+  // new theme is enabled, the UUIDs will still work and be the same.
```
Comment (sherakama) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Thanks for the comment details.

Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: woops. thats actually an old comment. they arent doing that

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 2]
```diff
@@ -283,3 +283,36 @@ function _su_humsci_profile_update_fix_menuposition_sections(array $sections) {
...
+    }
+  }
+
+  // Disable theme switcher module since we no longer need it.
+  \Drupal::service('module_installer')->uninstall(['role_theme_switcher']);
+  \Drupal::service('theme_installer')->uninstall(['material_admin']);
+
+  // Remove the role that is no longer necessary.
+  if ($role = Role::load('seven_admin_theme_user')) {
+    $role->delete();
```
Comment (sherakama) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install:316]: Does this also remove the roles from user entities? Looking at the class it does not appear to: https://api.drupal.org/api/drupal/core%21modules%21user%21src%21Entity%21Role.php/class/Role/8.7.x

File: docroot/themes/humsci/su_humsci_admin/scss/base/base.scss [hunk 3]
```diff
@@ -1,165 +0,0 @@
...
-// Overrides for the Matertial Design Admin theme.
-//
-
-@import
-  'decanter-no-markup';
-
-// Alerts.
-.messages.messages--warning,
-.messages.messages--warning .card,
-.toast.messages--warning,
```
Comment (sherakama) [inline docroot/themes/humsci/su_humsci_admin/scss/base/base.scss:14]: This is toast 😆!

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss [hunk 4]
```diff
@@ -0,0 +1,18 @@
+.layout-region-node-footer {
```
Comment (sherakama) [inline docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss:3]: missing charset at the top of the SCSS file. `@charset 'UTF-8';`

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
+.content-header {
```
Comment (sherakama) [inline docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss:3]: missing charset statement.

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  background: transparent;
+  text-align: center;
+}
+
+.breadcrumb {
+  display: none;
```
Comment (sherakama) [inline docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss]: Is there a better way to hide this? Can you not remove it from the layout?

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 1 above)
Comment (sherakama) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Damn, that seemed magical.

Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: well its actually not even needed. i went one approach of changing which admin theme was set. that was complicated. so i just change the base theme of this. much easier

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss (hunk 4 above)
Comment (pookmish) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss:3]: can you explain why this is needed?

File: docroot/themes/humsci/su_humsci_admin/scss/base/base.scss (hunk 3 above)
Comment (pookmish) [reply docroot/themes/humsci/su_humsci_admin/scss/base/base.scss:14]: https://youtu.be/rCQ_i13hg1o

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 2 above)
Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install:316]: It does in some way. Upon deleting the role if i load a user entity, the user does not have the `seven_admin_theme_user` role. To double check, i manually created the role again. the user still did not have the role.

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss (hunk 6 above)
Comment (pookmish) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss]: its the most simple way. all that it contains is a link to the home page. which the logo does anyways.

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss (hunk 4 above)
Comment (sherakama) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss:3]: Best practices for the most part:
* https://developer.mozilla.org/en-US/docs/Web/CSS/@charset

For SASS, it tells the compiler what to use for character encoding.

Comment (pookmish) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss:3]: makes sense for compiling. but the charset doesnt get compiled. not sure if thats neeed

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss (hunk 6 above)
Comment (sherakama) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss]: Does this introduce any accessibility issues? Off the top of my head, I don't think so but it does introduce a DX issue. I prefer to remove the breadcrumb via config or template as that is the most obvious place the next developer will look.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 2 above)
Comment (sherakama) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install:316]: This is great. Thank you for checking that out.

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss (hunk 4 above)
Comment (sherakama) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss:3]: https://www.sassmeister.com/gist/9537678

If I remove the `content` part the charset goes away. I suspect the compiler adds it when needed.

Comment (pookmish) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.node_form.scss:3]: oh thats interesting.

File: docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss (hunk 6 above)
Comment (pookmish) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss]: Ok, i removed the css and add an update hook to add visibility to the block.

Comment (sherakama) [reply docroot/themes/humsci/su_humsci_admin/scss/humsci_admin.portal.scss]: Thank you.

**Review by sherakama — APPROVED** [review]

**Pull request merged**